app/models/voucher.rb

```diff
@@ -1,8 +1,16 @@
 # frozen_string_literal: false
 
 class Voucher < ApplicationRecord
+  acts_as_paranoid
+
   belongs_to :enterprise
 
+  has_many :adjustments,
+           as: :originator,
+           class_name: 'Spree::Adjustment',
+           inverse_of: :voucher,
+           dependent: :nullify
```

Comment:
Since we are using `acts_as_paranoid`, do we want to allow a real destruction? If we don't then we could remove the `dependent` option and I think then the destroy would fail if it's referenced.